app/controllers/users_controller.rb

```diff
@@ -0,0 +1,65 @@
+class UsersController < ApplicationController
+  skip_before_filter :require_login, only: [:new, :create]
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+
+  # GET /users
+  def index
+    @users = User.all
+  end
+
+  # GET /users/1
+  def show
+  end
+
+  # GET /users/new
+  def new
+    @user = User.new
+  end
+
+  # GET /users/1/edit
+  def edit
+  end
+
+  # POST /users
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      auto_login(@user)
+      redirect_to root_path, notice: "User was successfully created."
+    else
+      render action: 'new'
+    end
+  end
+
+  # PATCH/PUT /users/1
+  def update
+    respond_to do |format|
+      if @user.update(user_params)
+        redirect_to @user, notice: 'User was successfully updated.'
+      else
+        render action: 'edit'
+      end
+    end
+  end
+
+  # DELETE /users/1
+  def destroy
+    @user.destroy
+    respond_to do |format|
+      redirect_to root_path
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_user
+      @user = User.find(params[:id])
+    end
+
+    # Never trust parameters from the scary internet, only allow the white list through.
```

Comment:
Line is too long. [88/80]<br>Incorrect indentation detected (column 4 instead of 2).